rtsf-at-checkout-event-reconciler/events/pim_lookup.go

```diff
@@ -8,24 +8,11 @@ import (
 	"errors"
 	"io/ioutil"
 	"net/http"
-	"os"
 
-	"github.com/edgexfoundry/app-functions-sdk-go/appcontext"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
 )
 
-func productLookup(productID string, edgexcontext *appcontext.Context) (ProductDetails, error) {
-
-	appSettings := edgexcontext.Configuration.ApplicationSettings
-	if appSettings == nil {
-		edgexcontext.LoggingClient.Error("No application settings found")
-		os.Exit(-1)
-	}
-
-	productLookupEndpoint, ok := appSettings["ProductLookupEndpoint"]
-	if !ok {
-		edgexcontext.LoggingClient.Error("ProductLookupEndpoint application setting not found")
-		os.Exit(-1)
-	}
+func productLookup(productID string, lc logger.LoggingClient, productLookupEndpoint string) (ProductDetails, error) {
```

Comment:
should we verify that productLookupEndpoint string is not empty as the origin code kind of doing that checking from the map itself

Reply:
as discussed, main does a validation on config, thus  productLookupEndpoint will not be empty